app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -920,15 +920,40 @@ def configure_vnic_with_requested_data(name, requested_vnic, destination_vnic, n
     end
 
     def private_load_allowed_networks(vlans, uid_ems_cluster)
-      profiles = get_vnic_profiles_in_cluster(uid_ems_cluster)
+      profiles = local_get_vnic_profiles_in_cluster(uid_ems_cluster)
       profiles.each do |profile, profile_network|
-        vlans[profile.id] = "#{profile.name} (#{profile_network.name})"
+        vlans[profile.uid_ems] = "#{profile.name} (#{profile_network.name})"
       end
 
       vlans['<Empty>'] = _('<No Profile>')
       vlans['<Template>'] = _('<Use template nics>')
     end
 
+    def local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      cluster_profiles = {}
+      profiles = all_vnic_profiles_as_lans_in_ems(ext_management_system)
+      cluster_networks = all_networks_as_switches_in_cluster(uid_ems_cluster)
+      profiles.each do |p|
+        profile_network = cluster_networks.detect { |n| n.id == p.switch.id }
+        if profile_network
+          cluster_profiles[p] = profile_network
+        end
+      end
+      cluster_profiles
+    end
+
+    def all_networks_as_switches_in_cluster(uid_ems_cluster)
+      cluster = EmsCluster.find_by(:uid_ems => uid_ems_cluster)
+      hosts = cluster.hosts
+      host_switches = hosts.collect(&:host_switches).flatten.compact.uniq
+      host_switches.collect(&:switch).flatten.compact.uniq
+    end
+
+    def all_vnic_profiles_as_lans_in_ems(ems)
+      ManageIQ::Providers::InfraManager::DistributedVirtualSwitch.where(ems_id: ems.id)
+        .collect(&:lans).flatten.compact.uniq
```

Comment:
I feel like this could be a relation with a distinct, like `ems.distributed_virtual_lans` which goes through `ems.distributed_virtual_switches`

Reply:
Untested but something like
```
agrare@agrare-thinkpad:~/src/manageiq/manageiq$ git diff
diff --git a/app/models/manageiq/providers/infra_manager.rb b/app/models/manageiq/providers/infra_manager.rb
index 25718af8c5..dfc0b723e2 100644
--- a/app/models/manageiq/providers/infra_manager.rb
+++ b/app/models/manageiq/providers/infra_manager.rb
@@ -8,6 +8,7 @@ module ManageIQ::Providers
     include AvailabilityMixin
 
     has_many :distributed_virtual_switches, :dependent => :destroy, :foreign_key => :ems_id, :inverse_of => :ext_management_system
+    has_many :distributed_virtual_lans, -> { distinct }, :through => :distributed_virtual_switches
     has_many :host_virtual_switches, -> { distinct }, :through => :hosts
 
     has_many :host_hardwares,             :through => :hosts, :source => :hardware
```